src/backend/protocols/go.imap/worker.go

```diff
@@ -84,35 +94,56 @@ func NewWorker(config WorkerConfig, id uint8) (worker *Worker, err error) {
 		}
 	}
 
-	// init Sender
 	return &w, nil
 }
 
 func (worker *Worker) Start() error {
 	var err error
-	(*worker).NatsSubs[0], err = worker.NatsConn.QueueSubscribe(worker.Config.NatsTopicFetcher, worker.Config.NatsQueue, worker.natsMsgHandler)
-	if err != nil {
-		return err
-	}
-	(*worker).NatsSubs[1], err = worker.NatsConn.QueueSubscribe(worker.Config.NatsTopicSender, worker.Config.NatsQueue, worker.natsMsgHandler)
+	(*worker).NatsSubs[0], err = worker.NatsConn.QueueSubscribe(worker.Config.NatsTopicSender, worker.Config.NatsQueue, worker.natsMsgHandler)
 	if err != nil {
 		return err
 	}
 	worker.NatsConn.Flush()
-	log.Infof("IMAP worker %d started", worker.Id)
+	log.Infof("IMAP worker %s started", worker.Id)
+
+	// start throttled jobs polling
+	for {
+		start := time.Now()
+		requestOrder := []byte(fmt.Sprintf(needJobOrderStr, worker.Id))
+		log.Infof("IMAP worker %s is requesting jobs to idpoller", worker.Id)
+		resp, err := worker.NatsConn.Request(worker.Config.NatsTopicPoller, requestOrder, time.Minute)
+		if err != nil {
+			log.WithError(err).Warnf("[worker %s] failed to request pending jobs on nats", worker.Id)
+		} else {
+			worker.natsMsgHandler(resp)
+		}
+		// check for interrupt after job is finished
+		if worker.HaltGroup != nil {
+			worker.Stop()
+			break
+		}
+		elapsed := time.Now().Sub(start)
+		if elapsed < pollThrottling {
+			time.Sleep(pollThrottling - elapsed)
+		}
+	}
 	return nil
 }
 
+func (worker *Worker) Halt(wg *sync.WaitGroup) {
+
```

Comment:
empty func, why ?

Reply:
ghost func I forgot to remove ! Done in commit #4f474891